cloud-formation/s3-uploader.yaml

```diff
@@ -131,16 +148,65 @@ Resources:
       MinSize: 1
       DesiredCapacity: 1
       Tags:
-      - { Key: App, Value: s3-uploader, PropagateAtLaunch: true }
-      - { Key: Stack, Value: media-service, PropagateAtLaunch: true }
+      - { Key: App, Value: {Ref: App},  PropagateAtLaunch: true }
+      - { Key: Stack, Value: {Ref: Stack}, PropagateAtLaunch: true }
       - { Key: Stage, Value: {Ref: Stage}, PropagateAtLaunch: true }
 
-
   LaunchConfig:
     Type: AWS::AutoScaling::LaunchConfiguration
+    Metadata:
+      AWS::CloudFormation::Authentication:
+        S3AccessCreds:
+          type: S3
+          roleName:
+            Ref: Role
+          buckets:
+            - media-service-dist
+      AWS::CloudFormation::Init:
+        config:
+          users:
+            s3-uploader:
+              homeDir: "/home/s3-uploader"
+          files:
+            "/home/media-service/s3-uploader.deb":
+              source:
+                Fn::Join:
+                  - ""
+                  -
+                    - "https://s3-eu-west-1.amazonaws.com/media-service-dist/media-service/"
+                    - Ref: "Stage"
+                    - "/s3-uploader/s3-uploader.deb"
+            "/home/media-service/s3-uploader/conf/application.conf":
+              source:
+                Fn::Join:
```

Comment:
minor: [`Fn::Sub`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html#w2ab2c21c28c56c11) might be more readable here (probably a separate PR)